app/client/consent.tsx

```diff
@@ -34,4 +34,10 @@ const run = (): void => {
   }
 };
 
-run();
+if (document.readyState !== 'loading') {
+  run();
+} else {
+  document.addEventListener('DOMContentLoaded', (): void => {
```

Comment:
Good solution.

Reply:
I'd consider also adding a note about this to the comment at the top of the run function's definition.